Sources/Subprocess/Errors.swift

```diff
@@ -31,7 +31,8 @@ import Foundation
 public enum SubprocessError: Error {
 
     /// The process completed with a non-zero exit code
-    case exitedWithNonZeroStatus(Int32)
+    /// and a custom error message.
+    case exitedWithNonZeroStatus(Int32, String)
```

Comment:
I think adding this could be a breaking change? If so, thoughts on a better way to go about this for backward compatibility?

Reply:
This will produce a warning in any code that is grabbing the associated value out of `exitedWithNonZeroStatus`.  And depending on how they are using the associated value they could get further errors/warnings since the new associated value will be extracted as a tuple instead of a single Int32.

From a `switch` statement with `case existedWithNonZeroStatus(let val):` this is the warning
> Enum case 'exitedWithNonZeroStatus' has 2 associated values; matching them as a tuple is deprecated

So, yes, in general I would consider this a breaking change requiring a major version number bump.

If there are other breaking changes that are planned or underway then we could update all at once with all the breaking changes.

Reply:
Another option I could do is add a flag to the Shell class and use that boolean to either use the old error or the new one if the flag was set. 

@cyrusingraham @mm512 any thoughts about keeping backward compatibility for this or going with a breaking change?

Reply:
@watkyn I would keep things simple and go with a breaking change here. The message included in the error will be helpful and I don't think it will be difficult for most projects to jump to next version of Subprocess.